quickstarts/microsoft.web/web-app-asp-app-on-asev3-create/main.bicep

```diff
@@ -0,0 +1,165 @@
+@description('Optional. Add unique suffix string to ASE name.')
+param addUniqueSuffix bool = true
+
+@description('Required. Use existing virtual network and subnet.')
+param useExistingVnetandSubnet bool = false
+
+@description('Required. Resource Group name of virtual network if using existing vnet and subnet.')
+param vNetResourceGroupName string = 'rg-asev3-vnet'
+
+@description('Required. The Virtual Network (vNet) Name.')
+param virtualNetworkName string = 'vnet-asev3'
+
+@description('Required. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. An Array of 1 or more IP Address Prefixes for the Virtual Network.')
+param vNetAddressPrefixes array = [
+  '172.16.0.0/16'
+]
+
+@description('Required. The subnet Name of ASEv3.')
+param subnetAddressPrefix string = '172.19.0.0/24'
+
+@description('Required. The subnet Name of ASEv3.')
+param subnetName string = 'snet-asev3'
+
+@description('Required. The subnet properties.')
+param subnets array = [
+  {
+    name: 'snet-asev3'
+    addressPrefix: '172.16.0.0/24'
+    // @description('Required. Delegation name of the ASEv3 subnet.')
+    delegations: [
+      {
+        name: 'Microsoft.Web.hostingEnvironments'
+        properties: {
+          serviceName: 'Microsoft.Web/hostingEnvironments'
+        }
+      }
+    ]
+    privateEndpointNetworkPolicies: 'Enabled'
+    privateLinkServiceNetworkPolicies: 'Enabled'
+    networkSecurityGroupName: 'nsg-asev3'
+  }
+]
+
+@description('Required. Name of ASEv3.')
+param aseNamePrefix string = 'asev3-ilb'
+@description('Required. Dedicated host count of ASEv3.')
+param dedicatedHostCount string = '0'
+@description('Required. Zone redundant of ASEv3.')
+@allowed([
+  true
+  false
+])
+param zoneRedundant bool = false
+@description('Optional. Create a private DNS zone for ASEv3.')
+param createPrivateDNS bool = true
+@description('Required. Load balancer mode: 0-external load balancer, 3-internal load balancer for ASEv3.')
+@allowed([
+  0
+  3
+])
+param internalLoadBalancingMode int = 3
+
+@description('Required. Name of the Network Security Group.')
+@minLength(1)
+param networkSecurityGroupName string = 'nsg-asev3'
+
+@description('Required. Array of Security Rules to deploy to the Network Security Group.')
+param networkSecurityGroupSecurityRules array
+
+@description('Optional. It is only for unique string generation base on timestamp.')
+param timeStamp string = utcNow()
+
+// Variable definitions
+var uniStr = substring('${uniqueString(resourceGroup().id, timeStamp)}', 0, 4)
+var aseName = addUniqueSuffix ? '${aseNamePrefix}-${uniStr}' : aseNamePrefix
+var virtualNetworkId = useExistingVnetandSubnet ? resourceId(vNetResourceGroupName, 'Microsoft.Network/virtualNetworks', virtualNetworkName) : resourceId('Microsoft.Network/virtualNetworks', virtualNetworkName)
```

Comment:
There is a scenario 2 that virtual network can be place in another resource group. I need condition to deploy different resource group to support this.

- Scenario 1 : Deploy ASEv3 to a new virtual network in the same resource group
- Scenario 2 : Deploy ASEv3 to an existing virtual network in different resource group

Reply:
```resourceId(vNetResourceGroupName, 'Microsoft.Network/virtualNetworks', virtualNetworkName) ```

will work in all cases since the defaultValue is the current resourceGroup - same for the subnetId below